app/controllers/concerns/course_basic_detail_concern.rb

```diff
@@ -53,4 +54,28 @@ def build_course
       .find(params[:code])
       .first
   end
+
+  def get_previous_course_creation_params
+    return unless params.has_key?(:course)
+
+    @course_creation_params = params.require(:course).permit(
+      :qualification, :maths, :english, :science
+    )
+  end
+
+  def go_to_next_step(current_step:, course_params:)
```

Comment:
I reckon this could be a service - and would allow us to give a bunch of test cases decoupled from controller requests to get the various paths :eyes: